### ISSUE TITLE
How to derive the provision key from the platform key?

### ISSUE THREAD

**Issue by deliciousdish** [intel/SGXDataCenterAttestationPrimitives#233]
Hi, all. I have tried to create my own provisiong certificate service. 
Now i have succeed to set my Registeration Server ID and decrypt the platform key from the platform manifest. But how to derive the provision key from the platform key? I tried the simulation code and replace the BASE_PROVISION_KEY with platform key, but i didn't get the same provision key as the physical host. The question is:
1. Is the EGETKEY logic in the simulation same as the logic in the hardware, or they are different?
2. Is the platform key play the same role as the BASE_PROVISION_KEY in the simulation? 
3. Is there any docs about the provison key derivation logic or it's a business secret?
Thanks!!

**Comment by andyzyb** [comment]
I don't think you can decrypt the platform root key from manifest, that means anyone can decrypt it and it won't be secure.

**Comment by andyzyb** [comment]
There is no guarantee that simulation uses the same derivation process as hardware.

**Comment by hyjiang** [comment]
> If I don't want to be supported by Intel backend, how to derive PCK from platform root key. I have tired to set my Registeration Server ID and decrypt the platform root key from the platform manifest. But when I replace the BASE_PROVISION_KEY with the platform root key, I can't get the same PCK as the physical host(with the same cpu_svn, isv_svn, mrsigner and others, i am sure). So, that means BASE_PROVISION_KEY and platform root key did not play the same role in the derive process. Is there any public docs about the derive process?

The public doc @ https://www.intel.com/content/dam/www/public/us/en/documents/white-papers/supporting-intel-sgx-on-mulit-socket-platforms.pdf , but seems the derive algorithm is not public.

**Comment by jsun39** [comment]
> If I don't want to be supported by Intel backend, how to derive PCK from platform root key. I have tired to set my Registeration Server ID and decrypt the platform root key from the platform manifest. But when I replace the BASE_PROVISION_KEY with the platform root key, I can't get the same PCK as the physical host(with the same cpu_svn, isv_svn, mrsigner and others, i am sure). So, that means BASE_PROVISION_KEY and platform root key did not play the same role in the derive process. Is there any public docs about the derive process?

I am not sure whether I get your point.  If you don't want to user Intel backend server, of course you can setup your owner server, but your still need to import the PRK/PCK materials from Intel in practice.  Of course if you just want to go through the process(using your owner backend server), there are other ways to do it.

**Comment by bgotowal** [comment]
Closing due to inactivity.

Clarified that simulation and hardware key derivation are not guaranteed to be identical, and that PCK derivation details are not fully public, with references shared in the thread. If you still need help, please reopen or file a new issue with your exact target flow and current DCAP versions.
